### PR TITLE
Added BVTs and integration with AppVeyor

### PR DESCRIPTION
**Pull request by anmenaga**

### CODE REVIEW

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+
+            $metadataXmlPath = Join-Path $script:TestSourceRoot "metadata.xml"
+            $metadataXml = Get-Content $metadataXmlPath
+        }
+    
+        function Get-MockCmdlet {[CmdletBinding()] param()
+            return $PSCmdlet
+        }
+        
+        It "Checks type coversion to CLR types" {
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: typo "conversion"

Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: I see this in other places, check the whole script

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+        It "Checks parsing metadata" {
+            
+            $tmpcmdlet = Get-MockCmdlet
+            $result = ParseMetadata -metadataXml $metadataXml -metaDataUri 'https://SomeUri.org' -cmdletAdapter 'ODataAdapter' -callerPSCmdlet $tmpcmdlet
+            $result.Namespace | Should Be "ODataDemo"
+            $result.DefaultEntityContainerName | Should Be "DemoService"
+
+            
+            $result.EntitySets.Length | Should Be 7
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Can you make the comparison value dynamic?  Retrieve it from a known source.  The magic numbers may change in the future.

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+            $rx = new-object System.Text.RegularExpressions.Regex('\bNestedModules = @\([^\)]*\)', ([System.Text.RegularExpressions.RegexOptions]::IgnoreCase -bor [System.Text.RegularExpressions.RegexOptions]::Singleline))
+            $nestedModules = $rx.Match($fileContents).Value;
+
+            $rx2 = new-object System.Text.RegularExpressions.Regex('([\w]*\.cdxml)', ([System.Text.RegularExpressions.RegexOptions]::IgnoreCase -bor [System.Text.RegularExpressions.RegexOptions]::Singleline))
+            $rx2.Matches($nestedModules).Count | Should Be 8
+        }
+    }
+
+     Context "OData v4 validation test cases" {
+
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: extra whitespace?

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        It "Verifies that generated module has correct contents" {
+            
+            $GlobalMetadata = New-Object System.Collections.ArrayList
+            $metadata = ParseMetadata -MetadataXML $metadatav4Xml -MetadataSet $GlobalMetadata
+            $GlobalMetadata.Add($metadata)
+            $normalizedNamespaces = @{}
+            $entitySet = $metadata.EntitySets[0]
+            [string]$generatedModuleName = $entitySet.Type.Name
+            $moduleDir = join-path $TestDrive "v4Module"
+            mkdir $moduleDir
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: did you run this through script-analyzer?  shouldn't use aliases in a script

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+            # based on Redfish Schema DSP8010 / 2016.1 / 31 May 2016
+
+            try { ExportODataEndpointProxy } catch {} # calling this here just to initialize module variables
+
+            foreach($metaXml in $metaXmls)
+            {
+                ParseMetadata -MetadataXML $metaXml -ODataVersion '4.0' -MetadataUri 'http://fakeuri/redfish/v1/$metadata' -Uri 'http://fakeuri/redfish/v1'
+            }
+
+            @($GlobalMetadata | ?{if ($_) {$_}}).Count | Should Be 159
```
Comment (SteveL-MSFT) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: same as previous, try to get the verification values dynamically if the metadata changes, if this is coming from the xml can you use an xpath query to get the expected result?

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 (hunk 2 above)
Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Updated.

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 (hunk 3 above)
Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Updated.

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 (hunk 4 above)
Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Updated.

File: Tests/Pester.PowerShell.ODataUtils.Tests.ps1 (hunk 5 above)
Comment (anmenaga) [inline Tests/Pester.PowerShell.ODataUtils.Tests.ps1]: Updated.

**Pull request merged**